script/SingletonDeployer_v1.s.sol

```diff
@@ -0,0 +1,275 @@
+//--------------------------------------------------------------------------
+
+contract SingletonDeployer_v1 is ProtocolConstants_v1 {
+    using Strings for string;
+
+    IDeterministicFactory_v1 public factory =
+        IDeterministicFactory_v1(deterministicFactory);
+
+    bytes32 factorySalt = keccak256(abi.encodePacked("inverter-deployment"));
+
+    //External contracts
+
+    address public ext_InverterReverter_v1;
+    address public ext_Governor_v1;
+    address public ext_TransactionForwarder_v1;
+    address public ext_FeeManager_v1;
+
+    //@note Add IssuanceToken here?
+
+    //Factories
+    address public fac_ModuleFactory_v1;
+    address public fac_OrchestratorFactory_v1;
+
+    //Modules
+
+    //Authorizer
+    address public mod_Aut_Roles_v1;
+    address public mod_Aut_TokenGated_Roles_v1;
+    address public mod_Aut_EXT_VotingRoles_v1;
+
+    //Funding Managers
+    address public mod_FM_BC_Bancor_Redeeming_VirtualSupply_v1;
+    address public mod_FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1;
+    address public mod_FM_Rebasing_v1; //@note Do we add this?
+    //address public mod_FM_DepositVault_v1;
+
+    //Logic Modules
+    address public mod_LM_PC_Bounties_v1;
+    address public mod_LM_PC_KPIRewarder_v1;
+    address public mod_LM_PC_PaymentRouter_v1;
+    address public mod_LM_PC_RecurringPayments_v1;
+    address public mod_LM_PC_Staking_v1;
+
+    //Payment Processors
+    address public mod_PP_Simple_v1;
+    address public mod_PP_Streaming_v1;
+
+    //Orchestrator
+    address public orc_Orchestrator_v1;
+
+    //--------------------------------------------------------------------------
+    // Factory Usage
+
+    // Deploy the Governor_v1.
+    /*
+    // Deploy without Arguments
+    bytes memory implBytecode = vm.getCode("Governor_v1.sol:Governor_v1");
+    govImplementation = factory.deployWithCreate2(salt, implBytecode);
+
+    // Deploy with Arguments
+    bytes memory proxyArgs = abi.encode(address(govImplementation), communityMultisig, bytes(""));
+    bytes memory proxyCode = vm.getCode("TransparentUpgradeableProxy.sol:TransparentUpgradeableProxy");
+    bytes memory proxyBytecode = abi.encodePacked(proxyCode, proxyArgs);
+    gov = Governor_v1(
+        address(
+                factory.deployWithCreate2(salt, proxyBytecode)
+        )
+    ); */
+
+    function createExternalSingletons() public {
+        console2.log(
+            "--------------------------------------------------------------------------------"
+        );
+        console2.log("Create External Singletons");
+        console2.log("-External Contracts");
+
+        //External contracts
+        ext_InverterReverter_v1 = factory.deployWithCreate2( //@note no Beacon structure here right?
+        factorySalt, vm.getCode("InverterReverte_v1.sol:InverterReverter_v1")); //@note can we add a string parameter for the factory? So we can log which contract is called here?
```

Comment:
If not maybe add a log before?  "---------\n Deploying Governor implementation:" and then the output of the factory.